census_extractomatic/api.py

```diff
@@ -2009,6 +2021,303 @@ def table_geo_comparison_rowcount(table_id):
+
+        # 'C' table with iterations, choosing 'A' iteration, e.g., C10001A
+        if 'C' + tabulation_code + 'A' in tables:
+            return 'C' + tabulation_code + 'A'
+
+        # 'B' table with iterations, choosing 'A' iteration, e.g., B10001A
+        if 'B' + tabulation_code + 'A' in tables:
+            return 'B' + tabulation_code + 'A'
+
+        else:
+            return ''
+
+    def process_result(row):
+        """ Converts a SQLAlchemy RowProxy to a dictionary. 
+
+        params: row - row object returned from a query
+        return: dictionary with either profile or table attributes """
+
+        row = dict(row)
+
+        if row['type'] == 'profile':
+            result = {
+                'type': 'profile',
+                'full_geoid': row['full_geoid'],
+                'full_name': row['display_name'],
+                'sumlevel': row['sumlevel'],
+                'sumlevel_name': row['sumlevel_name'] if row['sumlevel_name'] else '',
+                'url': build_profile_url(row['display_name'], row['full_geoid']),
+                'relevance': row['relevance'] #TODO remove this
+            }
+
+        if row['type'] == 'table':
+            table_id = choose_table(row['tables'].split())
+
+            result = {
+                'type': 'table',
+                'table_id': table_id,
+                'tabulation_code': row['tabulation_code'],
+                'table_name': row['table_title'],
+                'simple_table_name': row['simple_table_title'],
+                'topics': row['topics'].split(', '),
+                'unique_key': row['tabulation_code'],
+                'subtables': row['tables'].split(),
+                'url': build_table_url(table_id),
+                'relevance': row['relevance'] #TODO remove this
+
+            }
+
+        if row['type'] == 'topic':
+            result = {
+                'type': 'topic',
+                'topic_name': row['topic_name'],
+                'url': row['url'],
+                'relevance': row['relevance'] #TODO remove this
+            }
+
+        return result
+
+    def slugify(name):
+        ''' Slugifies a string by (1) removing non-alphanumeric / space 
+        characters, (2) converting to lowercase, (3) turning spaces to dashes
+
+        params: name - string to change
+        return: slugified string 
+        '''
+
+        name = re.sub('[^0-9a-zA-Z ]', '', name)
+        name = name.lower()
+        return name.replace(' ', '-')
+
+    def build_profile_url(display_name, full_geoid):
+        ''' Builds the censusreporter URL out of name and geoid.
+        Format: https://censusreporter.org/profiles/full_geoid-display_name/"
+
+        >>> build_profile_url("Columbus, IN Metro Area", "31000US18020")
+        "https://censusreporter.org/profiles/31000US18020-columbus-in-metro-area"
+
+        '''
+
+        new_name = slugify(display_name)
+        return "https://censusreporter.org/profiles/" + full_geoid + "-" + new_name + "/"
```

Comment:
Ah, I see why we're slugify-ing. Instead of replicating the slugify code in here (where it's possible that it would get out of sync with the code in the Django app), let's just send the URL `https://censusreporter.org/profiles/{full_geoid}`. The Django app will forward the user to the correct URL with the slugified display name.